scripts/generate-md-exports.mjs

```diff
@@ -63,40 +80,54 @@ async function createWork() {
   console.log(`📄 Converting ${numFiles} files with ${numWorkers} workers...`);
 
   const selfPath = fileURLToPath(import.meta.url);
-  const workerPromises = new Array(numWorkers - 1).fill(null).map((_, idx) => {
+  const workerPromises = new Array(numWorkers - 1).fill(null).map((_, id) => {
     return new Promise((resolve, reject) => {
-      const worker = new Worker(selfPath, {workerData: workerTasks[idx]});
-      let hasErrors = false;
-      worker.on('message', data => {
-        if (data.failedTasks.length === 0) {
-          console.log(`✅ Worker[${idx}]: ${data.success} files successfully.`);
-        } else {
-          hasErrors = true;
-          console.error(`❌ Worker[${idx}]: ${data.failedTasks.length} files failed:`);
-          console.error(data.failedTasks);
-        }
+      const worker = new Worker(selfPath, {
+        workerData: {id, noCache, cacheDir: CACHE_DIR, tasks: workerTasks[id]},
       });
+      let hasErrors = false;
+      worker.on('message', taskFinishHandler);
       worker.on('error', reject);
       worker.on('exit', code => {
         if (code !== 0) {
-          reject(new Error(`Worker[${idx}] stopped with exit code ${code}`));
+          reject(new Error(`Worker[${id}] stopped with exit code ${code}`));
         } else {
-          hasErrors ? reject(new Error(`Worker[${idx}] had some errors.`)) : resolve();
+          hasErrors ? reject(new Error(`Worker[${id}] had some errors.`)) : resolve();
         }
       });
     });
   });
   // The main thread can also process tasks -- That's 65% more bullet per bullet! -Cave Johnson
-  workerPromises.push(processTaskList(workerTasks[workerTasks.length - 1]));
+  workerPromises.push(
+    processTaskList({
+      noCache,
+      cacheDir: CACHE_DIR,
+      tasks: workerTasks[workerTasks.length - 1],
+      id: workerTasks.length - 1,
+    }).then(taskFinishHandler)
+  );
 
   await Promise.all(workerPromises);
 
   console.log(`📄 Generated ${numFiles} markdown files from HTML.`);
   console.log('✅ Markdown export generation complete!');
 }
 
-async function genMDFromHTML(source, target) {
+const md5 = data => createHash('md5').update(data).digest('hex');
+
+async function genMDFromHTML(source, target, {cacheDir, noCache}) {
   const text = await readFile(source, {encoding: 'utf8'});
+  const hash = md5(text);
+  const cacheFile = path.join(cacheDir, hash);
```

Comment:
We generate a lot of markdown files right? I wonder if there are issues caching them all in a single directory. Maybe we should split them up to be in different folders?

Reply:
It's about 8.5k files. I think we're fine up to 10k but I'll split if I notice a slow down